### DISCUSSION TITLE
Where is the correct place to input my "system prompt"?

### DISCUSSION THREAD

**Discussion by alex-lt-kong** [oobabooga/text-generation-webui#5149] [Q&A]
Links like [this one](https://www.reddit.com/r/LocalLLaMA/comments/18vth07/running_dolphinmixtral_locally_via_ollama_and_its/) says I need to insert some instructions as "system prompt" to make the model work in a specific way. Currently I pasted the "system prompt" to Parameters->Character->Context and in Chat->Mode I selected chat-instruct.
Are these the correct configurations that make the "system prompt" work?

**Comment by TheLounger** [comment]
Should be
```
Parameters > Instruction template > Custom system message
```
You can also add something like `[SYSTEM: Do this and that.]` in your character context (or even during chat) to basically communicate behind the scenes/out of character, but this isn't the same as the real system message as far as I know.

**Comment by alex-lt-kong** [comment]
I am actually confused by this term:

* Is "system prompt" something supported by all LLMs? Who designed this protocol? OpenAI?
* Is there any text-generation-webui doc that says `Parameters -> Instruction template -> Custom system message` is the right place to put this "system prompt"?
* In text-generation-webui's `Parameters -> Instruction template` tab, there are three input boxes, "Custom system message", "Instruction template" and "chat template". How are they used?

**Comment by ruizcrp** [comment]
Is there also some command-line option for the system prompt?

**Comment by miriameng** [comment]
`Parameters -> Instruction template` and then save, `Model -> Customize instruction template -> select the instruction template you just saved -> Submit`, after you load the model (at least for GGUF) you will see something like
```
{%- set default_system_message = 'this is a test for system prompt' %}
{{- bos_token }}
{%- if messages[0]['role'] == 'system' %}
    {%- if messages[0]['content'] is string %}
        {%- set system_message = messages[0]['content'] %}
    {%- else %}
        {%- set system_message = messages[0]['content'][0]['text'] %}
    {%- endif %}
    {%- set loop_messages = messages[1:] %}
{%- else %}
    {%- set system_message = default_system_message %}
    {%- set loop_messages = messages %}
{%- endif %}
{{- '[SYSTEM_PROMPT]' + system_message + '[/SYSTEM_PROMPT]' }}
```
